source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -271,6 +280,21 @@ namespace
                 const RtToken portType(elem->getType());
                 RtValue::fromString(portType, valueStr, input->getValue());
             }
+            const string& colorSpace = elem->getColorSpace();
```

Comment:
This fixes colorspace, unit and unittype on inputs. As noted this does not address any missing attributes on nodes. (colorspace). There are no units on nodes so not looking at it here.